### PR TITLE
Further changes for Auth0 react 

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR builds on #25 and supersedes it. The changes in this PR depend on #29, so please compare changes with the branch of that PR.

This PR replaces the deprecated use-auth0-hooks library with the @auth0-react library. The main change besides renaming is how the auth0 `accessToken` is obtained, and it is by calling `getAccessTokenSilently` where applicable.

### CODE REVIEW

File: components/LayoutWithAuth0.js [hunk 1]
```diff
@@ -45,60 +68,70 @@ class LayoutWithAuth0 extends Component {
...
-    const { accessToken, isAuthenticated } = auth
-    const { isUserRequested } = this.state
+    const { auth0 } = this.props
 
-    if (isAuthenticated && accessToken && !isUserRequested) {
-      // Set a flag to prevent duplicate fetches while awaiting the calls below to return.
-      this.setState({ isUserRequested: true })
+    if (this.acccessTokenIsUnfetched()) {
+      const accessToken = await auth0.getAccessTokenSilently()
+      this.setState({ accessToken })
```
Comment (landonreed) [inline components/LayoutWithAuth0.js:120]: Were you able to verify if setting the access token in the state was necessary? I'm wondering whether the auth0 library also uses react state hooks to store it if already fetched.

File: components/NavBar.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 import { useRouter } from 'next/router'
-import { useAuth } from 'use-auth0-hooks'
 
 import NavLink from './NavLink'
 
 export default function NavBar (props) {
   const { pathname } = useRouter()
-  const { isAuthenticated, isLoading } = useAuth()
+  const { isAuthenticated, isLoading } = useAuth0()
+  // FIXME: Fetch adminUser?
```
Comment (landonreed) [inline components/NavBar.js:9]: This FIXME needs a bit more context.

**Review by landonreed — APPROVED** [review]
A couple of minor things but otherwise looks great! Thanks for handling and keeping things a bit simpler than the mess I started in #25.

File: components/LayoutWithAuth0.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply components/LayoutWithAuth0.js:120]: @landonreed Could you clarify? We are passing the token value around to `secureFetch` in the render method of the component, and it is going to be tricky to fit `async getAccessTokenSilently` in the render method.

File: components/NavBar.js (hunk 2 above)
Comment (binh-dam-ibigroup) [reply components/NavBar.js]: Fixed in https://github.com/ibi-group/otp-admin-ui/pull/30/commits/74071aafc6203b31aff0c8e777868107671e4506. Good catch. This was an experiment that went nowhere.

File: components/LayoutWithAuth0.js (hunk 1 above)
Comment (landonreed) [reply components/LayoutWithAuth0.js:120]: How we're handling right now is probably OK, but I think we should add a FIXME about using something a little more contained for securing our API calls with the access token. Maybe pointing to this or another example: https://github.com/auth0/auth0-react/blob/master/EXAMPLES.md#4-create-a-useapi-hook-for-accessing-protected-apis-with-an-access-token

Comment (binh-dam-ibigroup) [reply components/LayoutWithAuth0.js:122]: Noted in https://github.com/ibi-group/otp-admin-ui/pull/30/commits/d01a8a1c5e2894325ca8d5de21f1c729996ebd16.

Comment (binh-dam-ibigroup) [reply components/LayoutWithAuth0.js:122]: Further changes to access token management are now in #33, which removes `accessToken` from the state of the component.

File: components/LayoutWithAuth0.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
+import { withAuth0 } from '@auth0/auth0-react'
 import Head from 'next/head'
 import { withRouter } from 'next/router'
-import React, { Component } from 'react'
+import { Component } from 'react'
```
Comment (evansiroky) [inline components/LayoutWithAuth0.js:4]: Isn't the React import still needed in this file since there is jsx?

File: package.json [hunk 4]
```diff
@@ -7,9 +7,11 @@
     "build": "next build",
     "dev": "next",
     "deploy": "vercel",
-    "lint": "mastarm lint components pages --quiet"
+    "lint": "mastarm lint components pages --quiet",
+    "predev": "yarn install"
```
Comment (evansiroky) [inline package.json:11]: Is it possible to just say `yarn`?

**Review by evansiroky — APPROVED** [review]
Seems like it's on the right track. Approved pending merge conflict resolution and @landonreed approving.

File: components/LayoutWithAuth0.js (hunk 3 above)
Comment (binh-dam-ibigroup) [reply components/LayoutWithAuth0.js]: Good catch. Restored in https://github.com/ibi-group/otp-admin-ui/pull/30/commits/2ca33d46f23714e74e4f3c28d546279ded463ff2.

File: package.json (hunk 4 above)
Comment (binh-dam-ibigroup) [reply package.json]: Sure (https://github.com/ibi-group/otp-admin-ui/pull/30/commits/ea940355ef74829804443a392256fca253aeb68d)

**Comment by binh-dam-ibigroup** [comment]
I addressed PR comments, fixed merge conflicts, and fixed the lint-docs errors. I will merge on Jan 20 if no other concerns.

**Pull request merged**